### PR TITLE
Strain: Add canonical data

### LINKED ISSUE #588: strain: Implement canonical-data.json
We want to have a standard set of test inputs and outputs for each exercise to
make it easier to port them to new languages, as well as to help keep the
exercises in sync.

The Strain exercise can be found in the [./exercises/strain/][exercise-dir] directory.

The step-by-step instructions for how to compile a canonical-data.json file is
described in [this section of the contributing guide][contributing].

See http://exercism.io/contribute/canonical-data/strain for the up-to-date list of
language tracks that have an implementation of the Strain exercise.

The [tracking issue][tracking-issue] has more context.

[contributing]: https://github.com/exercism/x-common/blob/master/CONTRIBUTING.md#improving-consistency-by-extracting-shared-test-data
[tracking-issue]: https://github.com/exercism/x-common/issues/552
[exercise-dir]: https://github.com/exercism/x-common/tree/master/exercises/strain

### PR DESCRIPTION
**Pull request by natanaelsirqueira**
Following the discussion that happened in [this topic](https://forum.exercism.org/t/proposal-to-add-test-cases-to-the-strain-problem-spec/3961/8), this PR adds canonical data for the Strain exercise, using a list of test cases that have been implemented by the majority of the tracks.

Closes #588

### CODE REVIEW

**Comment by petertseng** [comment]
Please amend the commit message to include the string "closes https://github.com/exercism/problem-specifications/issues/588" or any equivalent string. Thank you.

This request is not affected by the fact that that issue is already closed. This commit should be marked as closing it anyway, even though it is already closed. That way all commits relevant to that issue remain accessible from that issue.

**Comment by petertseng** [comment]
Please amend the commit message to include the string "`closes https://github.com/exercism/problem-specifications/issues/588`" or any equivalent string. Thank you.

This request is not affected by the fact that that issue is already closed. This commit should be marked as closing it anyway, even though it is already closed. That way all commits relevant to that issue remain accessible from that issue.

File: exercises/strain/canonical-data.json [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+          },
+          "expected": []
+        },
+        {
+          "uuid": "f535cb4d-e99b-472a-bd52-9fa0ffccf454",
+          "description": "keeps everything",
+          "property": "keep",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 1"
```
Comment (ErikSchierboom) [inline exercises/strain/canonical-data.json:23]: Maybe do:

```suggestion
            "predicate": "fn(x) -> true"
```

I think this makes it more straightforward to see that is keeps everything.

File: exercises/strain/canonical-data.json [hunk 2]
```diff
@@ -0,0 +1,155 @@
+{
+  "exercise": "strain",
+  "cases": [
+    {
+      "description": "filter list returning only values that satisfy the predicate function",
```
Comment (ErikSchierboom) [inline exercises/strain/canonical-data.json:5]: Where possible, we tend to try and not use nested test groups. In this case, you've used very readable descriptions for the test cases, so my preference would be to remove the nesting. Curious to hear what others think.

File: exercises/strain/canonical-data.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+          },
+          "expected": []
+        },
+        {
+          "uuid": "2f42f9bc-8e06-4afe-a222-051b5d8cd12a",
+          "description": "discards everything",
+          "property": "discard",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 1"
```
Comment (ErikSchierboom) [inline exercises/strain/canonical-data.json:98]: Similar suggestion as with the corresponding keep test case:

```suggestion
            "predicate": "fn(x) -> true"
```

File: exercises/strain/canonical-data.json [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+          },
+          "expected": []
+        },
+        {
+          "uuid": "ca990fdd-08c2-4f95-aa50-e0f5e1d6802b",
+          "description": "discards nothing",
+          "property": "discard",
+          "input": {
+            "list": [1, 3, 5],
+            "predicate": "fn(x) -> x % 2 == 0"
```
Comment (ErikSchierboom) [inline exercises/strain/canonical-data.json:108]: Similar suggestion as with the corresponding keep test case:

```suggestion
            "predicate": "fn(x) -> false"
```

**Review by ErikSchierboom — APPROVED** [review]
Love this! Excellent PR

**Comment by ErikSchierboom** [comment]
CC @exercism/reviewers

**Comment by ErikSchierboom** [comment]
CC @exercism/reviewers

**Review by IsaacG — APPROVED** [review]

**Pull request merged**